### PR TITLE
Redundant allocation when sending headerBytes to workers

### PR DESCRIPTION
**Pull request by kernel32uiuc**
According to [Buffer API](https://nodejs.org/api/buffer.html#static-method-bufferfromarray), Buffer.from() copies the passed buffer data onto a new Buffer instance if the argument is Uint8Array. When we send headerBytes to the worker, it is casted to Uint8Array and then we call Buffer.from(). It seems unnecessary and we can keep it in Buffer.

### CODE REVIEW

File: ironfish/src/workerPool/tasks/mineHeader.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { mineHeader } from '../../mining/mineHeader'
 export type MineHeaderRequest = {
   type: 'mineHeader'
   batchSize: number
-  headerBytesWithoutRandomness: Uint8Array
+  headerBytesWithoutRandomness: Buffer
```
Comment (dguenther) [inline ironfish/src/workerPool/tasks/mineHeader.ts:11]: NodeJS uses [object cloning](https://nodejs.org/api/worker_threads.html#considerations-when-cloning-objects-with-prototypes-classes-and-accessors) when transferring data to a worker thread, so `headerBytesWithoutRandomness` will end up as a `Uint8Array` when the MineHeaderRequest is serialized and sent to the worker.

So for this reason, I'd like to leave this type as `Uint8Array`. However, in `mining/mineHeader`, I think that type can be changed to `Uint8Array`, and then we can avoid the same allocation.

**Review by dguenther — CHANGES_REQUESTED** [review]

File: ironfish/src/workerPool/tasks/mineHeader.ts (hunk 1 above)
Comment (kernel32uiuc) [reply ironfish/src/workerPool/tasks/mineHeader.ts]: Thanks for the explanation. Done with the changes.

**Review by dguenther — APPROVED** [review]
👍 Thanks for the change! Retargeting to staging and merging.

**Comment by kernel32uiuc** [comment]
Is this PR also eligible to claim points for?

**Comment by dguenther** [comment]
Yep! If you could submit it through the Claim Points form on our website, we'll get around to adding points for it in the near future.

**Pull request merged**